url3426a97ee4_rtreffer_minidns_tgz-pJ8-de_measite_minidns_minidnsrepl_MiniDnsReplJ8/src/de/measite/minidns/dnssec/algorithms/AlgorithmMap.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2015-2016 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dnssec.algorithms;
+
+import de.measite.minidns.DNSSECConstants.DigestAlgorithm;
+import de.measite.minidns.DNSSECConstants.SignatureAlgorithm;
+import de.measite.minidns.dnssec.DNSSECValidatorInitializationException;
+import de.measite.minidns.dnssec.DigestCalculator;
+import de.measite.minidns.dnssec.SignatureVerifier;
+import de.measite.minidns.record.NSEC3.HashAlgorithm;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class AlgorithmMap {
+    private Logger LOGGER = Logger.getLogger(AlgorithmMap.class.getName());
+
+    public static final AlgorithmMap INSTANCE = new AlgorithmMap();
+
+    private final Map<DigestAlgorithm, DigestCalculator> dsDigestMap = new HashMap<>();
+    private final Map<SignatureAlgorithm, SignatureVerifier> signatureMap = new HashMap<>();
+    private final Map<HashAlgorithm, DigestCalculator> nsecDigestMap = new HashMap<>();
+
+    @SuppressWarnings("deprecation")
+    private AlgorithmMap() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 36-49 into a separate method.